pandas/core/arrays/datetimes.py

```diff
@@ -1480,6 +1626,22 @@ def sequence_to_dt64ns(data, dtype=None, copy=False,
     # if dtype has an embedded tz, capture it
     tz = validate_tz_from_dtype(dtype, tz)
 
+    if isinstance(data, (ABCSeries, ABCIndexClass)):
+        data = data._data
+
+    if isinstance(data, DatetimeArrayMixin):
```

Comment:
this is the only possiblity here?

Reply:
Can you clarify? We've unboxed Series / Index by this point.

The function continues on for when real parsing is needed.

Reply:
ok maybe comment that